api/serializers.py

```diff
@@ -35,8 +39,46 @@ class Meta:
         model = User
         fields = ("token", "username", "password")
 
-
 class VideoSerializer(serializers.ModelSerializer):
+    def create(self, validated_data):
+        data = self.context.get("data")
+        sender = self.context.get("user")
+
+        # receiver = User.objects.get(id=validated_data.pop('receiver'))
+        # Data is prepended by "data:video/mp4;base64,", need to remove those 22 characters
+        # Encode converts from string to bytes
+        data_bytes = data[22:].encode("ascii")
+        instance = self.Meta.model(**validated_data)
+
+        instance.sender = sender
+
+        # We need to do this before save because we use it for fname
+        instance.created_at = now()
+
+        # ok we need to construct a filename
+        fname = "%d_%d_%s" % (sender.id, instance.receiver.id, instance.created_at)
+
+        # ContentFile takes the data and creates a pseudo-file.
+        # See https://docs.djangoproject.com/en/3.1/ref/files/file/#the-contentfile-class
+        cf = ContentFile(data_bytes, name=fname)
+        instance.video_file.save(fname, cf)
+
+        instance.save()
+        return instance
+
+    def to_representation(self, instance):
+        rep = super().to_representation(instance)
+
+        video_file = instance.video_file
+        video_file.open()
+
+        # Raw base64 bytes need to be decoded to a string
+        # and the data needs the header
+        rep["video_file"] = "data:video/mp4;base64," + video_file.read().decode("UTF-8")

```

Comment:
The file is encoded as ascii above but decoded here as UTF-8. They should be compatible because base64 encoding doesn't use any characters outside of ascii but why have it UTF-8 here and ascii above? Also worth noting that `str.encode` and `str.decode` both use utf-8 by default so no need to specify.